### PR TITLE
[RzIL] Warn the user about `io.cache`

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

It took me again way to long to resolve a "bug" which simply was a disabled `io.cache` setting.

I hope this warning helps at least somewhat until we address https://github.com/rizinorg/rizin/issues/3996

**Test plan**

All green

**Closing issues**

...

### CODE REVIEW

File: librz/analysis/il/analysis_il.c [hunk 1]
```diff
@@ -90,13 +90,29 @@ RZ_API void rz_analysis_il_config_add_label(RZ_NONNULL RzAnalysisILConfig *cfg,
...
+}
+
 /**
  * Create and initialize an RzAnalysisILVM with the current arch/cpu/bits configuration and plugin
  * \p init_state_reg optional RzReg to take variable values from, unless the plugin overrides them using RzAnalysisILInitState
  * \return RzAnalysisRzil* a pointer to RzAnalysisILVM instance
  */
 RZ_API RZ_OWN RzAnalysisILVM *rz_analysis_il_vm_new(RzAnalysis *a, RZ_NULLABLE RzReg *init_state_reg) {
 	rz_return_val_if_fail(a, NULL);
+	check_io_cache_setting(a);
```
Comment (thestr4ng3r) [inline librz/analysis/il/analysis_il.c:115]: Better put this only in the command that calls it, `rz_analysis_il_vm_new()` may be used outside an interactive context.

**Review by thestr4ng3r — CHANGES_REQUESTED** [review]

**Comment by thestr4ng3r** [comment]
My 2 cents, I don't think it is a good idea to suggest users unconditionally to use io.cache since the correct way is to set up appropriate writable maps. This warning would show even if the user has set up these maps by themselves already, in which case enabling io.cache would be wrong.

**Comment by Rot127** [comment]
> This warning would show even if the user has set up these maps by themselves already,

Would it be fine for you if we add this to the note? Meaning that users can setup the maps themselves?

**Comment by Rot127** [comment]
-.- Forgot of cause about all the test we would need to update. This doesn't makes sense of cause.

**Pull request closed without merging**